x-pack/plugins/security_solution/public/detections/components/host_isolation/use_host_isolation_action.tsx

```diff
@@ -0,0 +1,101 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useMemo } from 'react';
+import type { TimelineEventsDetailsItem } from '../../../../common';
+import { isIsolationSupported } from '../../../../common/endpoint/service/host_isolation/utils';
+import { HostStatus } from '../../../../common/endpoint/types';
+import { useIsolationPrivileges } from '../../../common/hooks/endpoint/use_isolate_privileges';
+import { endpointAlertCheck } from '../../../common/utils/endpoint_alert_check';
+import { useHostIsolationStatus } from '../../containers/detection_engine/alerts/use_host_isolation_status';
+import { ISOLATE_HOST, UNISOLATE_HOST } from './translations';
+import { getFieldValue } from './helpers';
+
+interface UseHostIsolationActionProps {
+  closePopover: () => void;
+  detailsData: TimelineEventsDetailsItem[] | null;
+  isHostIsolationPanelOpen: boolean;
+  onAddIsolationStatusClick: (action: 'isolateHost' | 'unisolateHost') => void;
+}
+
+export const useHostIsolationAction = ({
+  closePopover,
+  detailsData,
+  isHostIsolationPanelOpen,
+  onAddIsolationStatusClick,
+}: UseHostIsolationActionProps) => {
+  const isEndpointAlert = useMemo(() => {
+    return endpointAlertCheck({ data: detailsData || [] });
+  }, [detailsData]);
+
+  const agentId = useMemo(
+    () => getFieldValue({ category: 'agent', field: 'agent.id' }, detailsData),
+    [detailsData]
+  );
+
+  const hostOsFamily = useMemo(
+    () => getFieldValue({ category: 'host', field: 'host.os.name' }, detailsData),
+    [detailsData]
+  );
+
+  const agentVersion = useMemo(
+    () => getFieldValue({ category: 'agent', field: 'agent.version' }, detailsData),
+    [detailsData]
+  );
+
+  const isolationSupported = isIsolationSupported({
+    osName: hostOsFamily,
+    version: agentVersion,
+  });
+
+  const {
+    loading: loadingHostIsolationStatus,
+    isIsolated: isolationStatus,
+    agentStatus,
+  } = useHostIsolationStatus({
+    agentId,
+  });
+
+  const { isAllowed: isIsolationAllowed } = useIsolationPrivileges();
+
+  const isolateHostHandler = useCallback(() => {
+    closePopover();
+    if (isolationStatus === false) {
+      onAddIsolationStatusClick('isolateHost');
+    } else {
+      onAddIsolationStatusClick('unisolateHost');
+    }
+  }, [closePopover, isolationStatus, onAddIsolationStatusClick]);
+
+  const isolateHostTitle = isolationStatus === false ? ISOLATE_HOST : UNISOLATE_HOST;
+
+  const hostIsolationAction = useMemo(
+    () =>
+      isIsolationAllowed &&
+      isEndpointAlert &&
+      isolationSupported &&
+      isHostIsolationPanelOpen === false
+        ? [
+            {
+              name: isolateHostTitle,
+              onClick: isolateHostHandler,
+              disabled: loadingHostIsolationStatus || agentStatus === HostStatus.UNENROLLED,
```

Comment:
I found a fix for why the `Isolate host` options always remain disabled after some digging. It looks like the `loadingHostIsolationStatus` is always true b/c of some code I wrote before that no longer works. If you could modify the `use_isolation_status.tsx` file with the below diff, the isolate host option should work again. Let me know if you want me to send you a patch file, otherwise i just copy and pasted the diff below!

```
index 6a40898d0a1..0631964660b 100644
--- a/x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_host_isolation_status.tsx
+++ b/x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_host_isolation_status.tsx
@@ -37,6 +37,7 @@ export const useHostIsolationStatus = ({
     const abortCtrl = new AbortController();
     // isMounted tracks if a component is mounted before changing state
     let isMounted = true;
+    setLoading(true);
     let fleetAgentId: string;
     const fetchData = async () => {
       try {
@@ -73,15 +74,10 @@ export const useHostIsolationStatus = ({
       }
     };

-    setLoading((prevState) => {
-      if (prevState) {
-        return prevState;
-      }
-      if (!isEmpty(agentId)) {
-        fetchData();
-      }
-      return true;
-    });
+    if (!isEmpty(agentId)) {
+      fetchData();
+    }
+
     return () => {
       // updates to show component is unmounted
       isMounted = false;
```